game-core/src/main/java/games/strategy/engine/chat/ChatMessageListener.java

```diff
@@ -8,5 +8,9 @@
 
   void addSlapMessage(String message, PlayerName from);
 
-  void addStatusMessage(String message, boolean joined);
+  void addSlapMessage(String message);
+
+  void addPlayerJoinedMessage(String message);
+
+  void addPlayerLeftMessage(String message);
```

Comment:
side-note, eventually we should follow up to avoid the kingdom of nouns problem, the "add.*Message" is noisy.

eg:
```
messageReceived
playerJoined
playerLeft
slap
```

Conveys same info as the 'add.*Message'. I don't really care if you change it now, but I'm nervous about the 'consistency' argument when we add another method and force it to fall into the kingdom of nouns naming anti-pattern.